### ISSUE TITLE
Bug  on rails console when process is suspended

### ISSUE THREAD

**Issue by rafaelfranca** [rails/spring#557]
_From @rderik on March 28, 2018 20:12_

### Steps to reproduce
Run `rails c` then suspend it by pressing [Ctrl+z]

### Expected behaviour
The user shoudl be able to suspend the rails console and then the user should be able to use the shell without any interference.

### Actual behaviour
The rails console doesn't seem to be suspended and it is "fighting" with the shell for the input and some input goes to the shell and some to the rails console so it breaks all the workflow. 

you can see a asciinema here: https://asciinema.org/a/pDMvNsyNHek77Ulyx6Z95VTuH

If I can be of any help let me know.

### System configuration
**Rails version**:
Rails 5.1.5
**Ruby version**:
ruby 2.4.1p111 (2017-03-22 revision 58053) [x86_64-darwin17]

_Copied from original issue: rails/rails#32367_

**Comment by TylerRick** [comment]
@rderik This looks like the same problem as #361.

Isn't that annoying?! It's been bugging me for years that that doesn't work properly.

But you're in luck, because the fix (#514) has already been merged! :fireworks: :tada: 

It is not released as RubyGem yet, but you can use the version from git, which seems to be working for me:
```ruby
group :development do                                                          
  gem 'spring', github: 'rails/spring'    
end
```

https://github.com/rails/spring/blob/master/CHANGELOG.md lists this under Next release:

>  Properly suspend and resume on ctrl-z TSTP and CONT (#361)

**Comment by SampsonCrowley** [comment]
OMG it's finally fixed in 2.1.0
